src/Gui/3Dconnexion/navlib/NavlibPivot.cpp

```diff
@@ -219,8 +245,24 @@ long NavlibInterface::SetHitDirection(const navlib::vector_t& direction)
 
 long NavlibInterface::SetHitLookFrom(const navlib::point_t& eye)
 {
-    ray.origin.setValue(eye.x, eye.y, eye.z);
-    return 0;
+    navlib::bool_t isPerspective;
+
+    GetIsViewPerspective(isPerspective);
+
+    if (isPerspective) {
+        ray.origin.setValue(eye.x, eye.y, eye.z);
+    }
+    else {
+        auto pCamera = getCamera<SoCamera*>();
+        if (pCamera == nullptr) {
+            return navlib::make_result_code(navlib::navlib_errc::no_data_available);
+        }
+
+        SbVec3f position = pCamera->position.getValue();
+        ray.origin = position + orthoNearDistance * ray.direction;
+
+        return 0;
+    }
```

Comment:
This looks like a bug. If `isPerspective` is `true` then there's no defined return value.
Shouldn't the `return 0` be placed outside the else block?
```suggestion
    }
    return 0;
```

Reply:
Yes, you are right. I have overlooked this.